### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by ignatius22**
This project is about creating the Model part of the MVC structured Ruby on Rails application. It doesn't contain front end works, so you will find only migration and model classes.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  validates :link, presence: true, length: { minimum: 5 }
+  validates :user_id, presence: true
```
Comment (maelfosso) [inline app/models/post.rb]: Use directly `:user` to validate instead of `:user_id`

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  validates :user_id, presence: true
+  validates :post_id, presence: true
```
Comment (maelfosso) [inline app/models/comment.rb]: Directy use `:user` and `:post`

File: db/migrate/20200727090506_create_posts.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreatePosts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :text
```
Comment (maelfosso) [inline db/migrate/20200727090506_create_posts.rb:7]: These fields, don't you validate them?

File: db/migrate/20200727124429_create_comments.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[5.2]
+  def change
+    create_table :comments do |t|
+      t.string :commenter
+      t.string :text
```
Comment (maelfosso) [inline db/migrate/20200727124429_create_comments.rb]: Why all these fields?? Please, re-read the requirements to know the required fields

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thanks for the effort put in this project.
You did a good job. 
I would like you to put a bit more effort to fix some issues to move forward.

Kindly read the comments left below and make the suggested changes. Once done, submit a new code review request
Thanks

Mael FOSSO

File: app/models/post.rb (hunk 1 above)
Comment (ignatius22) [reply app/models/post.rb]: fixed

File: app/models/comment.rb (hunk 2 above)
Comment (ignatius22) [reply app/models/comment.rb]: fixed

File: db/migrate/20200727090506_create_posts.rb (hunk 3 above)
Comment (ignatius22) [reply db/migrate/20200727090506_create_posts.rb:7]: they are generated migrations. by using "rails generate migration" command we add those fields to our schema, we wanted more understanding about how it works so we didn't run a scaffold for those models but generated them, we did all the validation on the models.

File: db/migrate/20200727124429_create_comments.rb (hunk 4 above)
Comment (ignatius22) [reply db/migrate/20200727124429_create_comments.rb]: they are generated migration, validations are only done on the models

File: app/models/post.rb (hunk 1 above)
Comment (ignatius22) [reply app/models/post.rb]: fixed

File: app/models/comment.rb (hunk 2 above)
Comment (ignatius22) [reply app/models/comment.rb]: fixed

File: db/migrate/20200727090506_create_posts.rb (hunk 3 above)
Comment (ignatius22) [reply db/migrate/20200727090506_create_posts.rb:7]: fixed

File: db/migrate/20200727124429_create_comments.rb (hunk 4 above)
Comment (ignatius22) [reply db/migrate/20200727124429_create_comments.rb]: fixed

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Iggy Sani & Godson Tendongze Zeingkeang**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with the fixes, you are almost there. Just missing a small step! 🏃 

# Required Changes 🏗️ 🐛 💡

- [ ] Make sure you have the GithubActions linter check turned on and passing all checks

- [ ] [OPTINAL] Try to validate if the email string is actually in the email format, you can use both regular expressions or rails inbuilt classes for that

- [ ] [OPTINAL] Validate the password so it's strong enough, can you make sure it's at least 6 characters, has letters and numbers and at least one symbol?

-----

 **Kindly fix the pointed out issues before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: However, you are one issue away from getting this project approved. Check :arrow_down: 

- **[REQUIRED]** All code linter checks should pass. :pray: Adjust your code so they do.

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by rOluochKe — APPROVED** [review]
Hello Team!
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next project.
- Happy coding!

**Pull request merged**